src/gui/torrentmodel.cpp

```diff
@@ -314,47 +298,53 @@ void TorrentModel::handleTorrentsUpdated()
 
 // Static functions
 
-QIcon getIconByState(BitTorrent::TorrentState state)
+static QString getIconId(BitTorrent::TorrentState state)
 {
     switch (state) {
     case BitTorrent::TorrentState::Downloading:
     case BitTorrent::TorrentState::ForcedDownloading:
     case BitTorrent::TorrentState::DownloadingMetadata:
-        return getDownloadingIcon();
+        return "downloading";
```

Comment:
What's the advantage? It's still going to then be copied to a new `QString` to be returned, so there's no efficiency advantage.

Returning a `QLatin1String` would be incorrect since the `const char *` used to construct it isn't copied - unless we made the literals constants (in which case we could return a `const QString &`). Hmm - could also do that for the colours below.